site/app/controllers/grading/ElectronicGraderController.php

```diff
@@ -1945,6 +1945,12 @@ public function ajaxSaveMark($gradeable_id) {
             return;
         }
 
+        // get the user
+        if ($this->core->getUser()->getGroup() === User::GROUP_STUDENT){
+            $this->core->getOutput()->renderJsonFail('Insufficient permissions for custom marks');
+            return;
+        }
```

Comment:
I think this will stop peers from being able to grade at all. What we actually want is for them to not be able to modify/add/remove custom marks.

The check that you want to add should be in `saveGradedComponent`, where `setComment` and `setPoints` are called for a graded component. Confusingly, I think that these calls represent the changing of a custom mark.

Reply:
This is in the ajaxSaveMark method, which is used when a limited access grader or above edits a mark (changing title, amount of points the mark is worth, etc.). This already contians a check to prevent students from editting the mark (line 1961). While this change doens't stop peers from being able to grade, Evan is correct that the check should be made in saveGradedComponent